packages/web-client/app/services/merchant-info.ts

```diff
@@ -67,6 +71,30 @@ export default class MerchantInfoService extends Service {
       did: info.data.attributes.did,
     };
   }
+
+  @task *checkMerchantSlugUniquenessTask(
+    params: CheckMerchantSlugUniquenessTaskParams
+  ): TaskGenerator<boolean> {
+    let response = yield fetch(
+      `${config.hubURL}/api/merchant-infos/validate-slug/${params.slug}`,
+      {
+        method: 'GET',
+        headers: {
+          Authorization: 'Bearer: ' + this.hubAuthentication.authToken,
+          Accept: 'application/vnd.api+json',
+          'Content-Type': 'application/vnd.api+json',
+        },
+      }
+    );
```

Comment:
We might need to do a try catch + clear the hub auth token if stuff fails here too. This pattern is now pretty common across the DApp, wonder if we can a good way to abstract it.

Reply:
ya could be something like a `fetch` wrapper in the hub service, akin to [this](https://github.com/hashicorp/nomad/blob/86fca8f513e6b2a0d9a6adaf2c473465516ec4f6/ui/app/services/token.js#L95-L101)